README.md

```diff
@@ -240,3 +241,129 @@ Destination collections path is /path/to/collections.
 ```
 python lsr_role2collection.py --src-path /path/to/role_group --dest-path /path/to/collections --tests-dest-path /path/to/test_dir --role myrole --src-owner linux-system-roles
 ```
+
+# release_collection.py
+
+This script is used to convert the roles to collection format using
+[lsr_role2collection.py](#lsr_role2collectionpy) and build the collection file
+for publishing.  It doesn't do the publishing, that must be done separately.
+
+The list of roles is specified by default in the file `collection_release.yml`.  The format of this file is a `dict` of
+role names.  Each role name is a dict which must specify the `ref` which
+is the git tag, branch, or commit hash specifying the commit in the role repo to use to build the collection from.  You can optionally specify the `org` (default: `linux-system-roles`) and the `repo` (default: role name).
+
+The other collection metadata comes from `galaxy.yml` - namespace, collection name, version, etc.  The script reads this metadata.
+
+The script reads the list of roles from `collection_release.yml`.  For each
+role, it will clone the repo from github to a local working directory (if there
+is no local clone) and update the local clone to the commit specified by `ref`.
+The script calls [lsr_role2collection.py](#lsr_role2collectionpy) to convert
+each role to collection format using the galaxy namespace name and collection
+name.  After all of the roles have been converted, the script uses
+`ansible-galaxy collection build` to build the collection package file suitable
+for publishing.
+
+To publish, use something like
+```
+ansible-galaxy collection publish /path/to/NAMESPACE-COLLECTION_NAME-VERSION.tar.gz
+```
+
+## Pre-requisites
+
+`git clone https://github.com/linux-system-roles/auto-maintenance [-b STABLE_TAG]`
+
+`STABLE_TAG` can be omitted to use `master`.
+
+Or, figure out some way to download the correct versions of
+[lsr_role2collection.py](#lsr_role2collectionpy), `galaxy.yml`,
+`collection_release.yml`, and `release_collection.py` from this repo.
+
+You will need to ensure that the information in `galaxy.yml` (particularly the
+`version:` field), and the information in `collection_release.yml` (particularly
+the `ref:` fields), are correct and up-to-date for the collection you want to
+build and publish.
+
+## collection_release.yml format
+```yaml
+ROLENAME:
+  ref: TAG_OR_HASH_OR_BRANCH
+  org: github-organization
+  repo: github-repo
+```
+Where `ROLENAME` is the name of the role as it will appear in the collection (under `NAMESPACE/COLLECTION_NAME/roles/`).  `ref` is the git tag (preferred), commit hash, or branch to use (in the format used as the argument to `git clone -b` or `git checkout`).  `org` is the github organization (default `linux-system-roles`).  The `repo` is the name of the repo under the github organization, and the default is the `ROLENAME`, so only use this if you need to specify a different name for the role in the collection.
```

Comment:
> The `repo` is the name of the repo under the github organization, and the default is the `ROLENAME`

Just out of curiosity, can we have a repo which name is not identical to `ROLENAME`?

I'm afraid if we do, lsr_role2collection would not work in that case. It expects the directory name in COLLECTION_SRC_PATH (e.g., ~/linux-system-roles) is `ROLENAME`...

Reply:
Yes.  We are planning to release the ansible-sshd role with out collection.  The upstream for this is https://github.com/willshersystems/ansible-sshd.  It is included in the collection under the name `sshd` - https://galaxy.ansible.com/fedora/linux_system_roles
The script clones the repo under the name `sshd` using `git clone https://github.com/willshersystems/ansible-sshd /path/to/sshd` so that the `lsr_role2collection.py` script only sees the `sshd` name.

Reply:
I see. This is a good example.
```
sshd:
  org: willshersystems
  repo: ansible-sshd
  ref: v0.12.0
```
To convert sshd to the collection format by `lsr_role2collection`, we have to clone `sshd` as
```
git clone https://github.com/willshersystems/ansible-sshd sshd
```

Reply:
The script does it here https://github.com/linux-system-roles/auto-maintenance/pull/30/files#diff-a7a7d93dac329fce31220c456b0b9cf11c19bfbe2fc0f9482e8c1366821760e7R56-R68 - the final argument `roledir` in this case ends with `/sshd`